types/pouchdb-mapreduce/index.d.ts

```diff
@@ -94,17 +94,32 @@ declare namespace PouchDB {
          * Invoke a map/reduce function, which allows you to perform more complex queries
          * on PouchDB than what you get with allDocs().
          */
-        query<Model>(fun: string | Filter | ((doc: any) => void), opts: Query.Options, callback: (err: Core.Error, res: Query.Response<Content & Model>) => void): void;
+        query<Model>(fun: string | ((doc: Content & Model) => void), opts: Query.Options, callback: (err: Core.Error, res: Query.Response<Content & Model>) => void): void;
```

Comment:
Why aren't we using the Callback type here? Also, the fact that `reduce` is optional, wouldn't it make it so that we could use `Filter` everywhere in this `Query` namespace? If there's no reduce function or no explicit Reduction type declaration from the user, then Filter will work just as the inline function we see here.